src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -75,7 +75,7 @@ object SignedTransaction {
 
     for {
       key <- recoveredPublicKey
-      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
+      addrBytes = crypto.kec256(key.tail).slice(FirstByteOfAddress, LastByteOfAddress)
```

Comment:
As much as I can read rust, parity doesn't seem to be doing this: https://github.com/paritytech/parity/blob/master/ethkey/src/keypair.rs#L23-L28

I'd rather suspect the `key` is not in fact the public key having an extra byte. For now, could you add TODO comment reminding to investigate it further?